agera/src/main/java/com/google/android/agera/RepositoryCompilerStates.java

```diff
@@ -469,6 +470,33 @@
     RConfig<TVal> onConcurrentUpdate(@RepositoryConfig int concurrentUpdateConfig);
 
     /**
+     * Specifies a {@code disposer} to handle any intermediate values discarded from the data
+     * processing flow. Intermediate values may be discarded due to the rest of the flow being
+     * skipped by a skip directive or a termination clause ({@code thenSkip()}, {@code orSkip()} and
+     * {@code orEnd()}), or cancelled in compliance with the deactivation or concurrent update
+     * configuration. The disposer may want to check if the values should be disposed of in a proper
+     * way, such as calling {@link Closeable#close()} on all {@link Closeable} values.
+     *
+     * <p>In each occasion where an intermediate value is discarded, the disposer will receive the
+     * value computed by the last completed directive. If the last directive was an attempt that
+     * failed, the disposer will receive the failed {@link Result}.
+     *
+     * <p>The compiled repository will only avoid sending the current repository value (accessible
```

Comment:
This whole block assumes that the same disposable instance is sent multiple times from an operation to the repository. Although I think the described behaviour is the correct one, I would assume this would probably mean the operation has failed to properly pass the ownership of the disposable, most likely leading to all sorts of trouble in other parts of the code.

It kind of make me wonder about the disposing and passing of ownership as a whole here. It does mean that clients using repo.get() should never retain that value. It does however also mean that since repo.get() can be called on any thread, there's actually no guarantee that the disposable object returned haven't been disposed of already in another thread, even if the reference is only kept on the stack?

Does this mean that disposers should only get intermediate values, and that retained values in a repository should never be considered as disposable (nor sent to the disposer)?
If so it does remove some of the issues with this block, and the comment about initial value.